infra/experimental/sanitizers/ExecSan/execSan.cpp

```diff
@@ -98,24 +180,101 @@ std::vector<std::byte> read_memory(pid_t pid, unsigned long long address, size_t
   return memory;
 }
 
-void inspect(pid_t pid, const user_regs_struct &regs) {
-  auto memory = read_memory(pid, regs.rdi, kTripWire.length());
-  if (memory.size() == 0) {
-    return;
+// Construct a string with the memory specified in a register.
+std::string read_register_string(pid_t pid, unsigned long reg, unsigned long length) {
```

Comment:
nit: rename to `read_string`. `register` is not entirely accurate, since it's actually taking in a an address, rather than a register.